configuration.ts

```diff
@@ -94,7 +96,14 @@ export class Configuration {
     this.username = param.username;
     this.password = param.password;
     this.accessToken = param.accessToken;
-    this.basePath = param.basePath;
+    this.baseOptions = {
+      // placing headers first allows users to overwrite if they want to include custom headers of their own
+      headers: {
+        'tilled-client-name': 'tilled-node',
+        'tilled-client-version': versionNum
+      },
+      ...param.baseOptions
+    };
     this.baseOptions = param.baseOptions;
```

Comment:
If a user specifies `baseOptions` it will overwrite what we've supplied here. You can't set `this.baseOptions` back to back.

Line 107 needs to be removed.

We need to ensure that we always set our own headers. You need to check to see if `param.baseOptions` has any headers and if it does, append ours to it, if `params.baseOptions` doesn't have any headers, then we need to add ours.


Let's make 100% sure we are always setting our 2 headers and not letting the customer change/overwrite them.

Reply:
@dpatton1992 @BMayhew FYI

Reply:
> If a user specifies `baseOptions` it will overwrite what we've supplied here. You can't set `this.baseOptions` back to back.
> 
> Line 107 needs to be removed.
> 
> We need to ensure that we always set our own headers. You need to check to see if `param.baseOptions` has any headers and if it does, append ours to it, if `params.baseOptions` doesn't have any headers, then we need to add ours.
> 
> Let's make 100% sure we are always setting our 2 headers and not letting the customer change/overwrite them.

@tposch baseOptions allows partners to manually set the [axios request config](https://axios-http.com/docs/req_config). Is there anything in there that we should expose to partners? I feel like `timeout` could be useful. I was back and forth about swapping those two lines which would overwrite any headers set by our partners, but couldn't we also do something like this to guarantee our headers are included?
```
this.baseOptions = {
      ...param.baseOptions,
      headers: {
        ...param.baseOptions.headers,
        'tilled-client-name': 'tilled-node',
        'tilled-client-version': version
      },
    };
```